sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -159,6 +165,10 @@ private[hive] class HiveMetastoreCatalog(sparkSession: SparkSession) extends Log
           Some(partitionSchema))
 
         val logicalRelation = cached.getOrElse {
+          val partitionSchemaColumnNames = partitionSchema.map(_.name.toLowerCase).toSet
+          val filteredMetastoreSchema = StructType(metastoreSchema
```

Comment:
You mean just rename ```filteredMetastoreSchema``` to ```metastoreDataSchema```? Filtering out the partition columns from the schema will still be required. I agree that ```metastoreDataSchema``` is a better name though.